src/Backend/Modules/MediaLibrary/Ajax/UploadMediaItem.php

```diff
@@ -0,0 +1,259 @@
+<?php
+
+namespace Backend\Modules\MediaLibrary\Ajax;
+
+use Backend\Core\Engine\Authentication as BackendAuthentication;
+use Backend\Core\Engine\Base\AjaxAction as BackendBaseAJAXAction;
+use Backend\Core\Language\Language;
+use Backend\Modules\MediaLibrary\Domain\MediaItem\Command\CreateMediaItemFromSource;
+use Backend\Modules\MediaLibrary\Domain\MediaFolder\MediaFolder;
+use Backend\Modules\MediaLibrary\Domain\MediaItem\Event\MediaItemCreated;
+use Backend\Modules\MediaLibrary\Domain\MediaItem\MediaItem;
+
+/**
+ * This AJAX-action is being used to upload new MediaItem items and save them into to the database.
+ */
+class UploadMediaItem extends BackendBaseAJAXAction
+{
+    // override existing media
+    const OVERRIDE_EXISTING = false;
+
+    /**
+     * Execute the action
+     */
+    public function execute()
+    {
+        parent::execute();
+
+        try {
+            // 60 minutes execution time
+            set_time_limit(60 * 60);
+        } catch (\Exception $e) {
+            // Do nothing
+        }
+
+        // increase memory limit
+        ini_set('memory_limit', '128M');
```

Comment:
what if the memory limit on that server is already higher than 128M then you will lower it. Not sure if I like this part being in there